src/inputs/TextField/TextField.test.tsx

```diff
@@ -0,0 +1,36 @@
+import { render, fireEvent } from '@testing-library/preact';
+import TextField from './TextField';
+import { h } from 'preact'
+import styles from './TextField.module.css'
+
+describe('TextField', () => {
+  const props = {
+    onChange: jest.fn(),
+    label: 'test',
+    name: 'test-name',
+    placeholder: 'test-placeholder',
+    id: 'test-id',
+  };
+
+  it('should call onChange handler when typing onto it', () => {
+    const { getByPlaceholderText } = render(<TextField {...props} />);
+    const input = getByPlaceholderText('test-placeholder');
+    fireEvent.change(input, { target: { value: 'new value' } });
+    expect(props.onChange).toHaveBeenCalled();
+  });
+
+  it('should not call onChange handler when TextField is disabled', () => {
+    const { getByPlaceholderText } = render(<TextField {...props} disabled />);
+    const input = getByPlaceholderText('test-placeholder');
+    fireEvent.change(input, { target: { value: 'new value' } });
+    expect(props.onChange).not.toHaveBeenCalled();
+  });
+
+  it('should have specific (depends on how you name it) className active when TextField is focused', () => {
+    const { getByPlaceholderText, getByLabelText } = render(<TextField {...props} />);
+    const input = getByPlaceholderText('test-placeholder');
+    fireEvent.focus(input);
+    const label = getByLabelText('test');
+    expect(label.classList.contains('labelActive')).toBe(true);
```

Comment:
**thought:** You won't test it this way as we are using CSS modules that change their classnames when compiling. The easiest way would be checking if input is being focused, eg:

```
  const { getByRole } = render(
      <TextField label="Name" id="name" onFocus={() => {}} />
  );

  const textfieldElement = getByRole("textbox");

  act(() => {
    textfieldElement.focus();
    expect(textfieldElement).toHaveFocus();
  });
```